src/astro/utils/load.py

```diff
@@ -58,14 +62,77 @@ def load_file_into_dataframe(
         elif filetype == FileType.JSON:
             dataframe = pd.read_json(stream, **kwargs)
         elif filetype == FileType.NDJSON:
-            dataframe = pd.read_json(stream, lines=True, **kwargs)
+            # dataframe = pd.read_json(stream, lines=True, **kwargs)
+            dataframe = flatten_ndjson(normalize_config, stream)
         elif filetype == FileType.PARQUET:
             dataframe = pd.read_parquet(stream, **kwargs)
         else:
             raise ValueError(f"Unable to load file '{filepath}' of type '{filetype}'")
         return dataframe
 
 
+def flatten_ndjson(
+    normalize_config: Union[None, dict], stream: io.TextIOWrapper
+) -> pd.DataFrame:
+    """
+    Flatten the nested ndjson/json.
+     :param normalize_config: parameters in dict format of pandas json_normalize() function.
+        https://pandas.pydata.org/docs/reference/api/pandas.json_normalize.html
+     :param stream: io.TextIOWrapper object for the file
+     :type normalize_config: dict
+     :type stream: io.TextIOWrapper
+     :return: return dataframe containing the loaded data
+     :rtype: `pandas.DataFrame`
+    """
+
+    normalize_config = normalize_config or {}
+
+    check_ndjson_config_delimiter(normalize_config=normalize_config)
+
+    df = None
+    rows = stream.readlines(DEFAULT_CHUNK_SIZE)
+    while len(rows) > 0:
+        if df is None:
+            df = pd.DataFrame(
+                pd.json_normalize([json.loads(row) for row in rows], **normalize_config)
+            )
+        else:
+            df = df.append(
+                pd.json_normalize([json.loads(row) for row in rows], **normalize_config)
+            )
+        rows = stream.readlines(DEFAULT_CHUNK_SIZE)
+    return df
+
+
+def check_ndjson_config_delimiter(normalize_config) -> dict:
+    """
+    Validate pandas json_normalize() parameter for databases, since default params result in
+     invalid column name. Default parameter result in the columns name containing '.' char.
+     :param normalize_config: :param normalize_config: config in dict format of pandas json_normalize() function.
+        https://pandas.pydata.org/docs/reference/api/pandas.json_normalize.html
+     :type normalize_config: dict
+     :return: return updated config
+     :rtype: `dict`
+    """
+
+    replacement = "_"
+    illegal_char = "."
+
+    meta_prefix = normalize_config.get("meta_prefix")
+    if meta_prefix and meta_prefix == illegal_char:
```

Comment:
What are we gaining by exposing this parameter to end-uses?

Reply:
if we rely on the Pandas method to flatten the NDJSON, we do expose the interface to the user.